sycl/include/syclcompat/memory.hpp

```diff
@@ -531,26 +531,39 @@ static inline void *malloc(size_t &pitch, size_t x, size_t y,
   return detail::malloc(pitch, x, y, 1, q);
 }
 
-/// free
+/// Wait on the queue \p q and free the memory \p ptr.
 /// \param ptr Point to free.
 /// \param q Queue to execute the free task.
 /// \returns no return value.
+static inline void wait_and_free(void *ptr,
+                                 sycl::queue q = get_default_queue()) {
+  q.wait();
```

Comment:
`get_current_device().queues_wait_and_throw();`

Reply:
Do we want to keep the `q.wait()` too? As a user, we probably expect `wait_and_free(ptr, q)` to synchronize `q`. In situations where `q` isn't from the current device, that wouldn't happen otherwise.

Reply:
Done